fspiop-api/documents/Tracing v1.0.md

```diff
@@ -0,0 +1,221 @@
+   > 3.3.2.1. Ensure _tracestate_ is less than 512 characters. If necessary remove the last list members until the 512 character limit has been reached <sup>[1](https://www.w3.org/TR/trace-context-1/#tracestate-limits)</sup>.
+
+### 3.4 Forwarding a Trace
+
+Vendors receiving trace request headers ([Table 1](#table-1)) do not need to participate but they MUST forward it in outgoing requests unmodified.
+
+Unmodified header propagation is typically implemented in pass-through services like proxies. As such, this behavior <sup>[1](https://www.w3.org/TR/trace-context-1/#mutating-the-traceparent-field)</sup> may also be implemented in a service which currently does not collect distributed tracing information.
+
+### 4. Examples
+
+#### 4.1. POST /transfers request being sent by FSP1 to FSP2 through a Mojaloop Switch
+
+This example will result in a single end-to-end trace graph being created from `FSP1` to `FSP2` being routed through a `Mojaloop Switch` for a  **POST /transfers** linking the resulting **PUT /transfers** callback from `FSP2` to `FSP1` via the `Mojaloop Switch`.
+
+<br/>
+4.1.1. `FSP1` sends a **POST /transfers** to a `Mojaloop Switch` which includes the following generated trace information:<br/><br/>
+
+> traceparent: `00`-`0af7651916cd43dd8448eb211c80319c`-`b7ad6b7169203331`-`01`<br/>
+> tracestate: `fsp1`=`t61rcWkgMzE`<br/>
+
+<br/>
+4.1.2. The `Mojaloop Switch` receives the **POST /transfers** and responds with a `HTTP 202` accepted. The `Mojaloop Switch` continuous the trace by mutating the trace information as follows, before forwarding the the **POST /transfers** to `FSP2`:<br/><br/>
+
+> traceparent: 00-0af7651916cd43dd8448eb211c80319c-`00f067aa0ba902b7`-01<br/>
+> tracestate: `moja`=`00f067aa0ba902b7`,fsp1=t61rcWkgMzE<br/>
+
+<br/>
+4.1.3. `FSP2` receives the request  and responds with a `HTTP 202` accepted. The funds are committed and `FSP2` sends a **PUT /transfers** callback including the following mutated trace information to the `Mojaloop Switch`:<br/><br/>
+
+> traceparent: 00-0af7651916cd43dd8448eb211c80319c-`b9c7c989f97918e1`-01<br/>
+> tracestate: `fsp2`=`ucfJifl5GOE`,moja=00f067aa0ba902b7,fsp1=t61rcWkgMzE<br/>
+
+<br/>
+4.1.4. The `Mojaloop Switch` receives the **PUT /transfers** and responds with a `HTTP 200`. The `Mojaloop Switch` will reconstuct the trace from a combination of the _traceparent_ and _tracestate_ headers, generating a new _parent-id_ from the span-id contained in the list-member `moja`=`00f067aa0ba902b7`. The trace information will be mutated as follows, before forwarding the the **PUT /transfers** back to `FSP1`:<br/><br/>
+
+> traceparent: 00-0af7651916cd43dd8448eb211c80319c-`53ce929d0e0e4736`-01<br/>
+> tracestate: `moja`=`53ce929d0e0e4736`,fsp2=ucfJifl5GOE,fsp1=t61rcWkgMzE<br/>
```

Comment:
Why does `moja=00f067aa0ba902b7` get dropped here?

Reply:
Because a new span is created.

Reply:
I have re-worked the first example. Let me know if is now more clear.